src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java/JavaLanguageLexer.g4

```diff
@@ -469,10 +470,8 @@ mode TextBlock;
 
 mode StringTemplate;
 
-    STRING_TEMPLATE_MID: StringFragment '\\' '{'
-                         -> pushMode(DEFAULT_MODE), type(STRING_TEMPLATE_MID);
+    STRING_TEMPLATE_CONTENT: StringFragment+;
```

Comment:
Empty "content" is nonsense, producing this token for static analysis purposes is worthless.